scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/protocol.kt

```diff
@@ -100,7 +101,9 @@ fun convertRequestToRawInputContents(request: V2Dataplane.ModelInferRequest): V2
                         .toList()
                 }.toByteArray()
             }
-            DataType.FP16, // as data is stored as FP32 we treat the same. Its unclear if Triton would handdle this correctly though
+            DataType.FP16 -> {  // as data is stored as FP32 we treat the same. Its unclear if Triton would handle this correctly though
+                throw UnsupportedOperationException("FP16 is not presently supported in Pipeline joins")
```

Comment:
:thought_balloon: I'm not sure if we should throw an exception here.  I have the nasty feeling this would kill the entire topology and possibly the dataflow engine as a whole.

We do need better error handling overall, so errors like this are non-fatal, so perhaps the idea is to prioritise that rather than not throwing exceptions in the first place?  In the meantime, it's potentially dangerous to throw this exception here instead of just dropping the data on the floor.